addons/udes_stock/models/stock_picking.py

```diff
@@ -1879,6 +1879,51 @@ def _get_suggested_location_by_products(self, move_line_ids, products=None):
 
         return suggested_locations
 
+    def _get_suggested_location_by_product_lot(self, move_line_ids, products=None):
```

Comment:
If the main difference is adding the lot to the search domain, I would suggest to merge product and product by lot into one with extra func.

Reply:
Unfortunately due to the way these functions are called is by looking for a function with the name ```_get_suggested_location_by```_by policy_ so we'd have to change quite a lot to get them to use a single function. I could potentially call ```_get_suggested_location_by_product``` from ```_get_suggested_location_by_product_lot``` and filter those locations but I think that would lead to a less efficient function as I would need to do some re-work (re-aquire the quants and filter)

Reply:
Added a new function called by both policies